lib/input/azure_queue_storage.go

```diff
@@ -0,0 +1,169 @@
+// +build !wasm
+
+package input
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-storage-queue-go/azqueue"
+	"github.com/Jeffail/benthos/v3/lib/message"
+
+	"github.com/Jeffail/benthos/v3/lib/input/reader"
+	"github.com/Jeffail/benthos/v3/lib/log"
+	"github.com/Jeffail/benthos/v3/lib/metrics"
+	"github.com/Jeffail/benthos/v3/lib/types"
+)
+
+// AzureQueueStorage is a benthos reader.Type implementation that reads messages
+// from an Azure Queue Storage container.
+type azureQueueStorage struct {
+	conf AzureQueueStorageConfig
+
+	queueURL *azqueue.QueueURL
+
+	log   log.Modular
+	stats metrics.Type
+}
+
+const (
+	azQueueEndpointExp  = "https://%s.queue.core.windows.net"
+	devQueueEndpointExp = "http://localhost:10001/%s"
+	azAccountName       = "accountname"
+	azAccountKey        = "accountkey"
+	devAccountName      = "devstoreaccount1"
+	devAccountKey       = "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw=="
+)
+
+// parseConnectionString extracts the credentials from the connection string.
+func parseConnectionString(input string) (storageAccount, storageAccessKey string, err error) {
```

Comment:
This logic around parsing azure connection strings is repeated all over azure outputs and inputs. Not sure where to put it in order to improve this not so DRY approach.

Reply:
There's a directly of packages for stuff like this growing here: https://github.com/Jeffail/benthos/tree/master/internal/service, I think a new `azure` package there is probably a good place for it.

Reply:
Great, I'll try to move this logic to a new azure package.

Reply:
Created the azure package to place the shared constants and functions used among azure inputs and outputs. Only put the related to Azure Queue Storage, for now, I think I can create a distinct PR to add the others (Blob and Table Storage) later.